src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala

```diff
@@ -51,14 +55,22 @@ class DagInstanceRepositoryImpl extends DagInstanceRepository {
     } yield ()).transactionally
   ).map(_ => (): Unit)
 
-  override def getNewActiveDags(idToFilter: Seq[Long], size: Int): Future[Seq[DagInstance]] = db.run(
-    dagInstanceTable.filter(ji =>
-      !ji.id.inSet(idToFilter) && ji.status.inSet(DagInstanceStatuses.finalStatuses)
-    ).take(size).result
-  )
+  def getDagsToRun(idToFilter: Seq[Long], size: Int)(implicit executionContext: ExecutionContext): Future[Seq[DagInstance]] = {
+    val prefilteredResult = db.run(
+      dagInstanceTable.filter { di =>
+        !di.workflowId.in(
+          dagInstanceTable.filter(_.id.inSet(idToFilter)).map(_.workflowId)
+        ) && di.status.inSet(DagInstanceStatuses.nonFinalStatuses)
```

Comment:
Why not just test for `DagInstanceStatuses.InQueue`? `DagInstance`s with status `Running` should probably already be excluded by `idsToFilter`?

Reply:
NonFinalStatuses are correct. Executor gets a task to execute (it can be: deploy, run wich is update status) so every time when it finishes one task it has to be queried again.